src/components/LibrariesList.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react";
+import { PanelGroup } from "react-bootstrap";
+import { Store, Reducer } from "redux";
+import { State } from "../reducers/index";
+import { LibrariesData, LibraryData } from "../interfaces";
+import { connect } from "react-redux";
+import ActionCreator from "../actions";
+import LibrariesListItem from "./LibrariesListItem";
+
+export interface LibrariesListStateProps {
+  libraries?: LibrariesData;
+}
+
+export interface LibrariesListOwnProps {
+  store?: Store<State>;
+  csrfToken?: string;
+}
+
+export interface LibrariesListDispatchProps {
+  fetchData: () => Promise<LibrariesData>;
+}
+
+export interface LibrariesListState {
+  activeKey: string;
+}
+
+export interface LibrariesListProps extends LibrariesListStateProps, LibrariesListOwnProps, LibrariesListDispatchProps {};
+
+export class LibrariesList extends React.Component<LibrariesListProps, LibrariesListState> {
+
+  constructor(props: LibrariesListProps) {
+    super(props);
+    this.state = { activeKey: "" };
+    this.isActive = this.isActive.bind(this);
+    this.select = this.select.bind(this);
+  }
+
+  isActive(library: LibraryData) {
+    return this.props.libraries.libraries.indexOf(library) === parseInt(this.state.activeKey);
+  }
+
+  select(idx: string) {
+    this.setState({ activeKey: idx});
+    this.props.fetchData();
+  }
+
+  render(): JSX.Element {
+    return(
+      <PanelGroup accordion activeKey={this.state.activeKey}>
+        { this.props.libraries &&
+          this.props.libraries.libraries.map(library =>
+            <LibrariesListItem
+              key={library.uuid}
+              idx={`${this.props.libraries.libraries.indexOf(library)}`}
+              library={library}
+              active={this.isActive(library)}
+              select={this.select}
```

Comment:
How does the index argument work?

Reply:
It's what tells the panel whether it should be open or closed; if panel's idx matches parent PanelGroup's activeKey, panel is open.

Reply:
But how does `idx` end up being passed into it since it's not on this line? Is something putting it in automatically?

Reply:
What do you mean about it not being on this line...?

Reply:
I would expect this line to be `select={() => this.select(idx)}` - I don't know how `select` is getting the index.

Reply:
Oh I see, the `ListItem` component is doing it. Sorry I hadn't gotten to that part yet. I think it would be cleaner if the `ListItem` didn't need to know where it was in the list.

Reply:
Update: it worked!!